### ISSUE TITLE
Fit camera to object in the 3D visualizer

### ISSUE THREAD

**Issue by cheton** [cheton/cnc.js#9]
Referring to
http://stackoverflow.com/questions/14614252/how-to-fit-camera-to-object
https://scottbyrns.atlassian.net/wiki/display/THREEJS/How+to+Fit+Camera+to+Object

**Comment by cheton** [comment]
Update:
[v0.8.1](https://github.com/cheton/cnc.js/releases/tag/v0.8.1): Set the 3D visualizer's rotation pivot point to the object's center position after rendering a G-code file.

**Comment by cheton** [comment]
Todo: Set the rotation pivot point in the 3D visualizer to the current position of the milling cutter.

**Comment by cheton** [comment]
Source: http://stackoverflow.com/questions/14614252/how-to-fit-camera-to-object

To match the object height with the visible height:
```js
fov = 2 * Math.atan(height / (2 * dist)) * ( 180 / Math.PI); // in degrees
```
To match the object width with the visible width:
```js
fov = 2 * Math.atan((width / aspect) / (2 * dist)) * (180 / Math.PI); // in degrees
```

The algorithm has been implemented in [v0.10.4](https://github.com/cheton/cnc.js/releases/tag/v0.10.4).
```js
const CAMERA_DEFAULT_FOV = 50;

// Fits camera to object
// @param {number} width The object width
// @param {number} height The object height
// @param {THREE.Vector3} [lookTarget] The object's top position which is nearest to the camera
fitCameraToObject(width, height, lookTarget) {
    console.assert(_.isNumber(width));
    console.assert(_.isNumber(height));
    console.assert(lookTarget instanceof THREE.Vector3);

    let v1 = (lookTarget instanceof THREE.Vector3) ? lookTarget : new THREE.Vector3(0, 0, 0);
    let v2 = new THREE.Vector3(
        this.camera.position.x,
        this.camera.position.y,
        this.camera.position.z
    );
    let dist = v1.distanceTo(v2); // the distance from the camera to the closest face of the object
    let aspect = this.camera.aspect; // the aspect ratio of the canvas (width / height)

    width = Number(width) || 0;
    height = Number(height) || 0;

    // Pick the largest camera field-of-view
    let fov = Math.max(
        2 * Math.atan(height / (2 * dist)) * (180 / Math.PI),
        2 * Math.atan((width / aspect) / (2 * dist)) * (180 / Math.PI)
    );
    fov = fov || CAMERA_DEFAULT_FOV;

    this.camera.fov = fov;
    this.camera.updateProjectionMatrix();
}
```

The usage example might look like so:
```js
let objectWidth = dimension.delta.x;
let objectHeight = dimension.delta.y;
let lookTarget = new THREE.Vector3(0, 0, dimension.max.z);

this.fitCameraToObject(objectWidth, objectHeight, lookTarget);
```
